copylot/hardware/lights/voltran/voltran.py

```diff
@@ -0,0 +1,516 @@
+        self.timeout :int= timeout
+
+        # Laser Specs
+        self.serial_number :str = devID
+        self.part_num: int = None
+        self.wavelength:int = None
+        self.laser_shape :str = None
+
+        # Properties
+        self._curr_power = None
+        self._ctrl_mode = None
+        self._delay = None
+        self._ext_power_ctrl = None
+        self._current_ctrl = None
+        self._toggle_emission = None
+        self._pulse_power = None
+        self._pulse_mode = None
+        self._max_power = None
+
+        self.connect()
+
+    def connect(self):
+        """
+        Establish the serial communication with the device in COM port
+        
+        Retrieves information about the laser including:
+        - serial number
+        - part number
+        - wavelength
+        - maximum power
+        - beam shape
+        """
+        try:
+            if self.port != None:
+                try:
+                    self.address = serial.Serial(
+                        port=self.port,
+                        baudrate=self.baudrate,
+                        bytesize=serial.EIGHTBITS,
+                        parity=serial.PARITY_NONE,
+                        stopbits=serial.STOPBITS_ONE,
+                        timeout=self.timeout,
+                    )
+                    # self._echo_off()
+                    self._identify_laser()
+                except:
+                    logger.debug(f"No Laser found in {self.port}")
+            elif self.serial_number != None:
+                ports = list_ports.comports()
+                for port in ports:
+                    try:
+                        self.port = port
+                        self.address = serial.Serial(
+                            port=self.port,
+                            baudrate=self.baudrate,
+                            bytesize=serial.EIGHTBITS,
+                            parity=serial.PARITY_NONE,
+                            stopbits=serial.STOPBITS_ONE,
+                            timeout=self.timeout,
+                        )
+                        self._identify_laser()
+                        if devSN == self.serial_number:
+                            logger.info(f"Connected {self.port}: Laser: {self.devID}")
+                        else:
+                            self.disconnect()
+                            raise Exception
+                    except:
+                        logger.debug(f"No laser found in {self.port}")
+        except:
+            raise RuntimeError("No laser found...")
+
+    def disconnect(self):
+        """
+        Disconnects the device
+        """
+        self.address.close()
+        self.address = None
+
+    def is_connected(self):
+        """
+        Check if device is connected to COM Port and return True/False
+
+        """
+        logger.debug(f'{self.port} is open')
+        return self.address.is_open
+
+    def write_cmd(self, cmd, value=None):
```

Comment:
should this method be protected, aka renamed to `_write_cmd`?

Reply:
Thank you for catching this! changed on[04cffcc](https://github.com/czbiohub/coPylot/pull/148/commits/04cffcc53f2925aff1613e78e1b87f70d1fb7497)